### PR TITLE
fix(debugging): handle disabled probes via RCM

### PR DESCRIPTION
**Pull request by P403n1x87**
This change fixes the handling of disabled configuration targets in dynamic instrumentation. Because of the currently broken RCM callback signature, the disable events were not handled correctly and caused an exception to be thrown during processing of the RCM event. We fix this by working around the problematic callback signature.

Testing is performed via the backend demo job monitoring, which is what allowed the issue to surface in the first place.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by shatzi — DISMISSED** [review]
Awesome fix.

File: ddtrace/debugging/_probe/remoteconfig.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def _update_probes_for_config(self, config_id, config):
         # type: (str, Any) -> None
         prev_probes = self._configs.get(config_id, {})  # type: Dict[str, Probe]
         next_probes = (
-            {probe.probe_id: probe for probe in get_probes(config_id, config)} if config is not None else {}
+            {probe.probe_id: probe for probe in get_probes(config_id, config)} if config not in (None, False) else {}
```
Comment (shatzi) [inline ddtrace/debugging/_probe/remoteconfig.py:239]: What the semantic difference between False and None? 
It look like a broken API for _update_probes_for_config callback.

**Review by shatzi — APPROVED** [review]

File: ddtrace/debugging/_probe/remoteconfig.py (hunk 1 above)
Comment (P403n1x87) [reply ddtrace/debugging/_probe/remoteconfig.py:239]: The typing of the expected signature of the callback mechanism in the RCM client is a bit funny at the moment. I believe `False` is being (ab)used as a sentinel object to indicate disabling of configuration items 😞 .

**Review by majorgreys — APPROVED** [review]

**Comment by mabdinur** [comment]
Do we need to backport this fix?

**Comment by P403n1x87** [comment]
> Do we need to backport this fix?

It might be good to have this on 1.10

**Pull request merged**